### PR TITLE
feat(mac-paste): normalize leading transcription capitalization before paste

### PR DESCRIPTION
**Pull request by macmixing**
- add mac-side paste capitalization heuristics that preserve sentence starts and lower Whisper-style default caps only when context proves the insertion is mid-sentence
- preserve dictionary-backed leading phrase casing during paste-time normalization with a dedicated mac dictionary casing store
- run capitalization normalization before existing spacing heuristics in PasteService and cover the ordering with paste execution tests
- add focused mac capitalization tests for sentence boundaries, fallback behavior, structural casing preservation, and dictionary-cased names
- update the Mac code map, engineering notes, README, and Xcode target wiring for the new paste-side capitalization layer

## Summary by CodeRabbit

* **New Features**
  * Improved paste handling: sentence-aware leading-capitalization normalization using focused context and previous non-whitespace character, with dictionary-backed preservation of intentional casing.

* **Documentation**
  * Updated docs to describe the new capitalization heuristics and dictionary-aware preprocessing in the paste workflow.

* **Tests**
  * Added unit tests and mocks covering capitalization heuristics and PasteService integration.

* **Chores**
  * Project/build updates to include new heuristic sources and tests.

### CODE REVIEW

File: Core/Services/Paste/Accessibility/PasteAXInspector.swift [hunk 1]
```diff
@@ -118,6 +127,56 @@ final class PasteAXInspector: PasteAXInspecting {
...
+    private func computePreviousNonWhitespaceCharacter(
+        element: AXUIElement,
+        caretLocation: Int
+    ) -> Character? {
+        guard caretLocation > 0 else { return nil }
+
+        if let value = valueString(for: element) {
+            let nsValue = value as NSString
+            guard caretLocation <= nsValue.length else { return nil }
+
```
[the remark this answers is not in the document]
Comment (macmixing) [reply Core/Services/Paste/Accessibility/PasteAXInspector.swift]: @coderabbitai you're suppposed to be on chill but are nitpicking the code too much
